app/Providers/FortifyServiceProvider.php

```diff
@@ -57,4 +63,19 @@ public function boot()
 
         RateLimiter::for('two-factor', fn (Request $request) => Limit::perMinute(5)->by($request->session()->get('login.id')));
     }
+
+    protected function patchRoutes(): void
+    {
+        if ($this->app->routesAreCached()) {
+            return;
+        }
+
+        $router = $this->app->make(Router::class);
+        $routes = $router->getRoutes();
+        collect(['create', 'store'])->each(function ($method) use ($routes) {
+            if ($route = $routes->getByAction(RegisteredUserController::class . '@' . $method)) {
+                $route->middleware('monica.signup_is_enabled');
+            }
+        });
+    }
```

Comment:
This should go directly in the `routes/web.php` file, there is no need to "patch" the route

Reply:
@asbiin, no, it shouldn't be in the `routes/web.php`.

Routes and controller are part of the `fortify` package. I don’t want to redefine the entire route, we just need to add our custom middleware, that’s why we alter the routes in the service provider.